src/Stage.java

```diff
@@ -62,26 +62,14 @@ public int getInstructionCount() {
         }
         return instructionCount;
```

Comment:
1. Please replace `int getInstructionCount()` method with `void setInstructionCount()` instead.
2. Create a new field `instructionCount` for `printState()` to use.